packages/components/src/components/hds/modal/index.ts

```diff
@@ -69,6 +69,18 @@ export default class HdsModal extends Component<HdsModalSignature> {
     super(owner, args);
 
     registerDestructor(this, (): void => {
+      // if the <dialog> is removed from the dom while open we emulate the close event
+      if (this._element && this._isOpen) {
+        this._element.dispatchEvent(new Event('close'));
```

Comment:
This might as well be `this.registerOnCloseCallback();` since that's what's being called in the end, but I found dispatching the event easier to read (as emulate the close action)